hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java

```diff
@@ -88,7 +92,8 @@ public synchronized void start(@NonNull final HederaState state, @NonNull final
         // If the latest immutable state has changed, we need to adjust the dispatcher and the
         // query-handler.
         if (!Objects.equals(state, lastUsedState)) {
-            dispatcher = new PreHandleDispatcher(state, servicesAccessor, context);
+            dispatcher =
+                    new PreHandleDispatcher(state, servicesAccessor, context, preHandleTxnAccessor);
```

Comment:
This does not work. If we really want to keep the class `PreHandleTxnAccessor`, we have to create a new instance when the state changes.

Reply:
Reverted it back and injected `PreHandleDispatcher` to `preHandle` methods for `Schedules`